prometheus_client/__init__.py

```diff
@@ -300,6 +304,73 @@ def _samples(self):
           ('_count', {}, self._count),
           ('_sum', {}, self._sum))
 
+def _floatToGoString(d):
+  if d == _INF:
+    return '+Inf'
+  elif d == _MINUS_INF:
+    return '-Inf'
+  else:
+    return repr(d)
+
+@_MetricWrapper
+class Histogram(object):
+  _type = 'histogram'
+  _reserved_labelnames = ['histogram']
+  def __init__(self, buckets=(.005, .01, .025, .05, .075, .1, .25, .5, .75, 1.0, 2.5, 5.0, 7.5, 10.0, _INF)):
+    self._sum = 0.0
+    self._lock = Lock()
+    buckets = [float (b) for b in buckets]
+    if buckets != sorted(buckets):
+      # This is probably an error on the part of the user,
+      # so raise rather than sorting for them.
+      raise ValueError('Buckets not in sorted order')
+    if not buckets:
+      raise ValueError('Must have at least one bucket')
```

Comment:
But what if that one bucket is _INF?

I'd add the check after adding the _INF and then make sure there are at least two buckets.